### ISSUE TITLE
pyDataProvider报错，单机CPU运行

### ISSUE THREAD

**Issue by pkuyym** [PaddlePaddle/Paddle#769]
重载了pyDataProviderBase.py，单机CPU运行报错误，log为：
> F1208 12:13:47.471302 25426 ClassRegistrar.h:63] Check failed: mapGet(type, creatorMap_, &creator) Unknown class type: py

**Comment by backyes** [comment]
@pkuyym 

请尽可能提供多的配置信息。

**Comment by pkuyym** [comment]
> ################################### Data Configuration ###################################
TrainData(PyData(files = "train.list", load_data_module = "pyDataProvider", load_data_object = "GeneralDataProvider"))

################################### Algorithm Configuration ###################################
Settings(learning_rate=0.4, learning_rate_decay_a=1e-05, learning_rate_decay_b=0.0, batch_size=400, algorithm='sgd')

################################### Network Configuration ###################################
Layer(type = "data", name = "input1", size = 153)
Layer(type = "data", name = "input2", size = 1)
Layer(inputs = [Input("input1", parameter_name = "_layer1.w")], name = "layer1", bias = Bias(parameter_name = "_layer1.bias"), active_type = "tanh", type = "fc", size = 128)
Layer(inputs = [Input("layer1", parameter_name = "_layer2.w")], name = "layer2", bias = Bias(parameter_name = "_layer2.bias"), active_type = "tanh", type = "fc", size = 128)
Layer(inputs = [Input("layer2", parameter_name = "_output.w")], name = "output", bias = Bias(parameter_name = "_output.bias"), active_type = "softmax", type = "fc", size = 2)
Layer(type = "data", name = "label", size = 1)
Layer(inputs = [Input("output"), Input("label"), Input("input2")], type = "multi-class-cross-entropy", name = "cost")
Layer(inputs = [Input("output"), Input("label"), Input("input2")], type = "auc-validation", name = "auc")
Inputs("input1", "input2", "label")
Outputs("cost")

**Comment by reyoung** [comment]
1、PyDataProvider 已经被废弃了，请使用PyDataProvider2。
2、这个原因似乎是编译时没有enable python解释器。使用cmake的时候，请开启`-DWITH_PYTHON=ON`

**Comment by pkuyym** [comment]
我使用的是paddle_api_cmd_1_0_0_7环境，请问能否自己编译paddle，然后替换其中的paddle_trainer？

**Comment by reyoung** [comment]
> 我使用的是paddle_api_cmd_1_0_0_7环境，请问能否自己编译paddle，然后替换其中的paddle_trainer？

直接升级一下二进制就好了。

**Comment by pkuyym** [comment]
还有一个问题，paddle_api_cmd_1_0_0_7这个应该仅仅是一个提交mpi任务的环境吧，应该不会过时吧，有需要就直接替换二进制？

**Comment by reyoung** [comment]
> 还有一个问题，paddle_api_cmd_1_0_0_7这个应该仅仅是一个提交mpi任务的环境吧，应该不会过时吧，有需要就直接替换二进制？

1、不全是。这个软件包会把这个软件包里的二进制放到mpi上执行。所以二进制会老的。
2、直接替换也不太行。因为公司内的mpi集群用的linux版本非常老。导致本地编译的不一定能直接放上去运行。请联系 王燕飞 @backyes 查询现在百度集群训练的软件包

**Comment by backyes** [comment]
@pkuyym  

deeplearning.baidu.com/doc
